src/main/java/com/mraof/minestuck/item/ScalemateItem.java

```diff
@@ -0,0 +1,71 @@
+package com.mraof.minestuck.item;
+
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.ActionResultType;
+import net.minecraft.util.Hand;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.world.World;
+import net.minecraftforge.common.Tags;
+
+public class ScalemateItem extends Item
+{
+    private Item newItem;
+    
+    public ScalemateItem(Properties properties)
+    {
+        super(properties);
+    }
+    
+    @Override
+    public ActionResult<ItemStack> onItemRightClick(World worldIn, PlayerEntity playerIn, Hand handIn)
+    {
+        Item offhandItem = playerIn.getHeldItemOffhand().getItem();
+        ItemStack mainhandItemStack = playerIn.getHeldItem(handIn);
+        
+        if(offhandItem.isIn(Tags.Items.DYES_RED))
```

Comment:
If you don't like the big if-else block, consider the alternative to have a static item-to-scalemate map that is initialised during common setup.